src/gui/nodes/OSCSenderNodeEditor.cpp

```diff
@@ -96,7 +96,9 @@ OSCSenderNodeEditor::~OSCSenderNodeEditor()
 }
 
 void OSCSenderNodeEditor::timerCallback() {
-    std::vector<OSCMessage> oscMessages = oscSenderNodePtr->getOscMessages();
+
+    const std::vector<OSCMessage>& oscMessages = oscSenderNodePtr->getOscMessages();
```

Comment:
I added `const` and `&` to avoid copying here, since the vector's been copied already in `getOscMessages`. Does this do what I think it should? :)

Reply:
I removed the reference - when I was "stress testing" the OSCSender, I ran into an malloc error `Pointer being freed was not allocated`. I believe that `&` was the issue.

Reply:
..Nope, I was still able to reproduce the malloc error without `&`. I'll see if I can use XCode's debug tools to find out where it's coming from.

Reply:
Can I see the stack trace?  I'm guessing that  the `vector<..>::clear()` happens while the thread is iterating over it.   You'll need a mutex i.e. `juce::CriticalSection` to sync access to the vector.

```c++
private:
...
CriticalSection theLock;
...
```
```c++
    .... OSC thread
     ScopedLock sl (theLock);
       while (iter1.getNextEvent (msg, frame))
        {
            OSCMessage oscMsg = Util::processMidiToOscMessage (msg);
            oscSender.send (oscMsg);

            if (oscMessagesToLog.size() > maxOscMessages)
                oscMessagesToLog.erase ( oscMessagesToLog.begin() );
            oscMessagesToLog.push_back ( oscMsg );
        }
```

```c++
    ScopedLock sl (theLock);
    std::vector<OSCMessage> copied = oscMessagesToLog;
    oscMessagesToLog.clear();
    return copied;
```

Reply:
Ahh, I see it. It *is* related to this part in `OSCSenderNodeEditor::timerCallback`. It calls `OSCSenderNode::getOscMessages`, which is doing:

```cpp
std::vector<OSCMessage> copied = oscMessagesToLog;
oscMessagesToLog.clear();
return copied;
```

That `clear()` is freeing OSCMessage and OSCArgument instances. So..when malloc says the pointer being freed was *not* allocated.. Ooh, right, the above section is not copying the contents, it's copying the pointers to the instances?

OK, so I need to actually copy the messages before clearing them. I'll see how that works.